src/Data/Aeson/GADT/TH.hs

```diff
@@ -54,7 +54,7 @@ pattern Some x = This x
 #endif
 
 -- Do not export this type family, it must remain empty. It's used as a way to trick GHC into not unifying certain type variables.
-type family Skolem :: k -> k
+data family Skolem :: k -> k
```

Comment:
I spent a couple hours adding traces and very confused on why a gazillion instances were being returned until I remembered about https://github.com/obsidiansystems/dependent-sum-template/pull/7
Shouldn't the chunk of code that's equal between the two libs be re-used instead?

https://github.com/obsidiansystems/dependent-sum-template/blob/846df3885404b3c343453a2aa786eb6433a346d5/src/Data/GADT/TH/Internal.hs#L38-L63
https://github.com/obsidiansystems/aeson-gadt-th/blob/71a315a4873c2875ad737a3320320849bdcf8a2a/src/Data/Aeson/GADT/TH.hs#L57-L82